### PR TITLE
api: ext-proc timeout, fail-open, backendRefs

### PR DESCRIPTION
**Pull request by guydc**
**What this PR does / why we need it**:
Support timeout, fail open mode and plural backend refs for external processor.
The implementation PR will:
- Remove singular BackendRef (no compatibility issue with released API versions).
- Make backendRefs mandatory. 

**Which issue(s) this PR fixes**:
xref: #3170

### CODE REVIEW

File: test/cel-validation/envoyextensionpolicy_test.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ package celvalidation
 import (
 	"context"
 	"fmt"
+	"k8s.io/utils/ptr"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
```
Comment (shawnh2) [inline test/cel-validation/envoyextensionpolicy_test.go:15]: sort the imports please

**Review by arkodg — APPROVED** [review]
🚀

File: api/v1alpha1/ext_proc_types.go [hunk 2]
```diff
@@ -14,8 +14,26 @@ import (
 //
 // ExtProc defines the configuration for External Processing filter.
 type ExtProc struct {
-	// Service defines the configuration of the external processing service
+	// BackendRef defines the configuration of the external processing service
```
Comment (arkodg) [inline api/v1alpha1/ext_proc_types.go:17]: plan on deleting `backendRef` in a follow up PR ?

Comment (arkodg) [reply api/v1alpha1/ext_proc_types.go:17]: nvm this is a API PR

**Review by shawnh2 — APPROVED** [review]
lgtm

**Pull request merged**